src/main/java/org/influxdb/impl/InfluxDBImpl.java

```diff
@@ -739,4 +739,28 @@ public void dropRetentionPolicy(final String rpName, final String database) {
     execute(this.influxDBService.postQuery(this.username, this.password,
         Query.encode(queryBuilder.toString())));
   }
+
+  private void setLogLevel() {
```

Comment:
- This method is being called every time an object is created. Can we call it just once, statically?
- IMO the name of this method is not following the JavaBean convention: `setXXX` usually is used to change the state and expects 1..* parameter(s). Can you rename it to something like `configureLogLevel`?
- I'm almost sure all this method can be replaced with something like:
```Java
String value = System.getProperty...
if (value != null) {
   // logLevel = ... don't change this property with this method as there is already another method changing it.
   // instead, call directly setLogLevel(...)
   setLogLevel(LogLevel.valueOf(value.toUpperCase()));
}
```

Reply:
>This method is being called every time an object is created. Can we call it just once, statically?

I dont think so, I think we should allow developer to change the property value on the fly (as you can see in the unit test) for debugging purpose, or its visibility may differ from context to context (custom SecurityManager is used)

>IMO the name of this method is not following the JavaBean convention: setXXX usually is used to change the state and expects 1..* parameter(s). Can you rename it to something like configureLogLevel?

For me the naming convention setXXX(...) and setXXX() are not semantically ambiguous. But rename it to initLogLevel anyway

>I'm almost sure all this method can be replaced with something like

partially agree and fix accordingly, except keeping switch case since you still have check for null or invalid property value (either by conditional structure like this or catching exception from valueOf(String))

Reply:
> I dont think so, I think we should allow developer to change the property value on the fly (as you can see in the unit test) for debugging purpose
So, you want to provide a way to the developer to change the loglevel of InfluxDB objects already instantiated in order to make the debug easier. Is this correct? If so, then you expect the developer to create on his side something like what you did on the test case: `System.setProperty`. How do you expect that the developer will change the loglevel while the application is running?
If I got your intentions correctly and what I said before is right then I suggest you to do the right way and use JMX.

Reply:
> For me the naming convention setXXX(...) and setXXX() are not semantically ambiguous. But rename it to initLogLevel anyway

Please check the Section "8.3 Design Patterns for Properties" from the JavaBean API specification: http://download.oracle.com/otn-pub/jcp/7224-javabeans-1.01-fr-spec-oth-JSpec/beans.101.pdf .

Reply:
> partially agree and fix accordingly, except keeping switch case since you still have check for null or invalid property value (either by conditional structure like this or catching exception from valueOf(String))
Ok., I don't think you got my point of view. Let me be more clear: If you have 100 different loglevels, would you still keep using a switch/case statement? AFAIK, your code can be replaced and "secured" with a try/catch with no collateral effects.

Reply:
>So, you want to provide a way to the developer to change the loglevel of InfluxDB objects already instantiated in order to make the debug easier. Is this correct? If so, then you expect the developer to create on his side something like what you did on the test case: System.setProperty. How do you expect that the developer will change the loglevel while the application is running?
If I got your intentions correctly and what I said before is right then I suggest you to do the right way and use JMX.

As far as I know JMX is a technology for managing and monitoring rather than debugging the application. For debugging you would like to use a more flexible but simple way, and use of a System property here would be better. Two use cases:
1/ You can pass the log level to JVM at booting up
2/ During your debugging session, you can manage to set the log level of the client which is going to created somewhere else (I do not reject you can use JMX but it will never have such a flexibility like evaluating a java statement to change that System property value)

Reply:
>Please check the Section "8.3 Design Patterns for Properties" from the JavaBean API specification: http://download.oracle.com/otn-pub/jcp/7224-javabeans-1.01-fr-spec-oth-JSpec/beans.101.pdf .

I've had this already, so initLogLevel() is ok for you or not ?

Reply:
>Ok., I don't think you got my point of view. Let me be more clear: If you have 100 different loglevels, would you still keep using a switch/case statement? AFAIK, your code can be replaced and "secured" with a try/catch with no collateral effects.

OK, got it, fixed as your comment (use of Enum.valueOf(,))
Tks